internal/controller/serving/reconcilers/tier_config.go

```diff
@@ -0,0 +1,208 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconcilers
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	kservev1alpha1 "github.com/kserve/kserve/pkg/apis/serving/v1alpha1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/opendatahub-io/odh-model-controller/internal/controller/resources"
+)
+
+const (
+	TierAnnotationKey      = "alpha.maas.opendatahub.io/tiers"
+	TierConfigMapName      = "tier-to-group-mapping"
+	DefaultTenantNamespace = "maas-api"
+	DefaultTenantName      = "maas-default-gateway"
+)
+
+// Tier represents a subscription tier with associated user groups and level.
+//
+// Level determines precedence, where higher values take precedence over lower values.
+// This can be needed in scenarios when users belong to multiple groups across different tiers.
+type Tier struct {
+	Name        string   `yaml:"name"`                  // Tier name (e.g., "free", "premium", "enterprise")
+	Description string   `yaml:"description,omitempty"` // Human-readable description
+	Groups      []string `yaml:"groups"`                // List of groups that belong to this tier
+	Level       int      `yaml:"level,omitempty"`       // Level for importance (higher wins)
+}
```

Comment:
If we export some pieces form `maas-billing` project we can remove this code and re-use.

This is copied from https://github.com/opendatahub-io/maas-billing/blob/4ca3983dd96375dda13fce4544ed39592ccce3bc/maas-api/internal/tier/types.go#L9